dashboard/app/assets/javascripts/progress.js

```diff
@@ -101,5 +108,12 @@ window.dashboard.progress = (function () {
     })));
   };
 
+  progress.renderCourseProgress = function (scriptData) {
+    $('.user-stats-block').prepend(React.renderToStaticMarkup(React.createElement(dashboard.CourseProgress, {
+      stages: scriptData.stages
+    })));
+    progress.populateProgress(scriptData.name);
```

Comment:
How difficult would it be to make this true React, instead of just using React as a templating language and renderingToStaticMarkup? i.e. are there reasons we cant move the logic from populateProgress into our components?